bundles/org.openhab.binding.tsmart/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,105 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="tsmart"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="tsmart">
+
+		<label>Tesla T-Smart</label>
+		<description>Tesla T-Smart Immersion Heater Thermostat</description>
+
+		<channels>
+			<channel id="power" typeId="system.power"/>
+			<channel id="temperature" typeId="temperature">
+				<label>Temperature</label>
+				<description>Measured temperature across the sensors</description>
+			</channel>
+			<channel id="temperature-high" typeId="temperature">
+				<label>High Temperature</label>
+				<description>Measured temperature on the High sensor</description>
+			</channel>
+			<channel id="temperature-low" typeId="temperature">
+				<label>Low Temperature</label>
+				<description>Measured temperature on the Low sensor</description>
+			</channel>
+			<channel id="setpoint" typeId="temperature-setpoint"/>
+			<channel id="mode" typeId="mode"/>
+			<channel id="smart-mode" typeId="smart-mode"/>
+			<channel id="relay" typeId="relay-state"/>
+		</channels>
+
+		<config-description>
+			<parameter name="hostname" type="text" required="true">
+				<context>network-address</context>
+				<label>IP Address</label>
+				<description>IP address of the device</description>
+			</parameter>
+			<parameter name="refreshInterval" type="integer" unit="s" min="1">
+				<label>Refresh Interval</label>
+				<description>Interval the device is polled in sec.</description>
+				<default>60</default>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<channel-type id="relay-state">
+		<item-type>Switch</item-type>
+		<label>Relay State</label>
+	</channel-type>
+
+	<channel-type id="temperature">
+		<item-type>Number:Temperature</item-type>
+		<label>Temperature</label>
+		<description>Temperature of water in cylinder sensed by thermostat</description>
+		<category>Temperature</category>
+		<tags>
+			<tag>Measurement</tag>
+			<tag>Temperature</tag>
+		</tags>
+		<state readOnly="true" pattern="%.0f %unit%"/>
+	</channel-type>
+
+	<channel-type id="temperature-setpoint">
+		<item-type>Number:Temperature</item-type>
+		<label>Setpoint</label>
+		<description>Setpoint currently set on device</description>
+		<category>Heating</category>
+		<tags>
+			<tag>Setpoint</tag>
+			<tag>Temperature</tag>
+		</tags>
+		<state min="15" max="70" step="5" readOnly="false" pattern="%.0f %unit%"/>
```

Comment:
No, it'll be celsius only. Is your suggestion that I should I set the pattern to celsius, and this will enforce the limit correctly for an item set to another unit? I've tried to follow all the recent enhancement to UoM but have to admit I hadn't thought about this scenario!

Reply:
No, my only point is that I don't think `min`, `max` and `step` will take unit into consideration, but tbh, I'm not really sure. You could probably easily test that by changing the item's unit to °F and trying to go below/above defined boundaries?

If in fact UoM is not considered, you would either have to provide them dynamically (which is somewhat more complex) or set the boundaries to the lowest/highest allowed value in any of those units. So max would be 158 because 70 °C is 158 °F. Probably it can be assumed that no one would configure this to be in Kelvin. 😉

Reply:
I'm not sure where to go with this, and don't have too much time to invest. This is what I know:
1. This is a water heater, there's a bigger problem that openhab being confused if we're looking at 15°F or 158°C :)
2. min/max/step in MainUI reflect the configured unit as you thought
3. I think the device is only sold in the UK so 90%+ of users will work in °C
4. Device resets to 15°C if I try to go below it (e.g. 35°F).
5. Device goes to 70°C if I try to go to negative °C, (e.g. 15°F) - suspect this is conversion to bytes, needs looking into.
6. Device can be set to 10°C, just not via the setpoint, only via travel mode.
7. Device doesn't need steps of 5°C, appears to respect a step of 0.1°C. (I did this as mirroring the companion app)
8. I looked at some other thermostats (Nest, Tado, Wiser). All configure a setpoint as %unit%, only one sets a step of 0.1, none have min/max.

I think I should do the following (~mirrors other bindings):
1. Code guards to prevent setting min / max temps outside 15-70°C and log error.
2. Remove min, max, step properties from Setpoint

Alternative is to set the pattern="%.0f °C" and explore how the REST API, MainUI and sitemaps behave if my item has unit of °F?